core/src/main/java/org/mskcc/cbio/portal/dao/DaoCancerStudy.java

```diff
@@ -481,14 +487,91 @@ public static void deleteCancerStudy(String cancerStudyStableId) throws DaoExcep
             JdbcUtil.closeAll(DaoCancerStudy.class, con, pstmt, rs);
         }
     }
+    
+    /**
+     * Calls deleteCancerStudyByCascade() if cancer study exists by stable id.
+     * @param cancerStudyStableId
+     * @throws DaoException 
+     */
+    public static void deleteCancerStudyByCascade(String cancerStudyStableId) throws DaoException {
+        CancerStudy study = getCancerStudyByStableId(cancerStudyStableId);
+        if (study != null) {
+            deleteCancerStudyByCascade(study.getInternalId());
+        }
+    }
 
     /**
+     * Deletes a cancer study by internal id from db with foreign key constraints enforced.
+     * @param internalCancerStudyId
+     * @throws DaoException 
+     */
+    private static void deleteCancerStudyByCascade(int internalCancerStudyId) throws DaoException {
+        Connection con = null;
+        PreparedStatement pstmt = null;
+        ResultSet rs = null;
+        try {
+            con = JdbcUtil.getDbConnection(DaoCancerStudy.class);
+            pstmt = con.prepareStatement("DELETE FROM cancer_study WHERE cancer_study_id = ?");
+            pstmt.setInt(1, internalCancerStudyId);
+            int rows = pstmt.executeUpdate();
+            // throw dao exception if no rows affected by update
+            if (rows == 0) {
+                throw new DaoException("deleteCancerStudyByCascade() failed: No rows affected");
+            }
+            deleteUnusedReferencesFromDb();
+            removeCancerStudyFromCache(internalCancerStudyId);
+        } catch (SQLException e) {
+            throw new DaoException(e);
+        } finally {
+            JdbcUtil.closeAll(DaoCancerStudy.class, con, pstmt, rs);
+        }
+        reCacheAll();
+        System.out.println("deleted study:\nID: "+internalCancerStudyId);
+    }
+    
+    /**
+     * Deletes unused references from database that may have been unique to 
+     * a single cancer study that no longer exists.
+     * 
+     * Tables:
+     *    - cna_event
+     *    - mutation_event
+     *    - gistic
+     * @throws DaoException 
+     */
+    private static void deleteUnusedReferencesFromDb() throws DaoException {
+        Connection con = null;
+        PreparedStatement pstmt = null;
+        ResultSet rs = null;
+        try {
+            con = JdbcUtil.getDbConnection(DaoCancerStudy.class);
+            
+            // sql statements to remove unused references in mutation_event, cna_event, 
+            // and gistic after a study is deleted. 
+            String[] sqls = {
```

Comment:
I thought we decided to leave this out for now and research a better solution? It's fine to include though.